### DISCUSSION TITLE
How to import environment variable file in the phase setup_build_environment?

### DISCUSSION THREAD

**Discussion by longlong10087** [spack/spack#25813] [General]
I added a software package called test. There is an environment variable file bashrc under the source code.
I nedd to import this file before compiling. 
I want to perform this operation in the setup_build_environment phase, but the source code package has not been decompressed yet. 
Is there any way to decompress it at this stage?

**Comment by cosmicexplorer** [comment]
This isn't a direct answer to your question, but it looks like we are discussing possible changes to the `setup_build_environment` phase in https://github.com/spack/spack/discussions/13926 and your use case may be covered there.
